### ISSUE TITLE
Resolution 3840x2160

### ISSUE THREAD

**Issue by NoNameTrademark** [zbee/d4-always_gatehall#1]
Hi,

Your script is really practical! Can you make a version that works for resolution 3840x2160 16:9 plz?

Thanks :)

**Comment by zbee** [comment]
I do not have this resolution available to me, so I cannot test it.
If you can test it and report what the problems are then I would be happy to adjust it though.

**Comment by NoNameTrademark** [comment]
Can you do it with this screen?

3840x2160 (16:9 Widescreen)
Font Scale: Small
Cursore Scale : Large
High Contrast Cursor: On
Cursor Color Picker: #FFFFFF
Brightness: Default (Middle)

<a href="https://ibb.co/NF22Vm4"><img src="https://i.ibb.co/NF22Vm4/carte4k.png" alt="carte4k" border="0"></a> <a href="https://ibb.co/7gLn327"><img src="https://i.ibb.co/7gLn327/cartefilterouvert4k.png" alt="cartefilterouvert4k" border="0"></a> <a href="https://ibb.co/hFQNsPM"><img src="https://i.ibb.co/hFQNsPM/carte-Journalouvert4k.png" alt="carte-Journalouvert4k" border="0"></a> <a href="https://ibb.co/PzF5BmP"><img src="https://i.ibb.co/PzF5BmP/Main4k.png" alt="Main4k" border="0"></a> <a href="https://ibb.co/VN9jxRK"><img src="https://i.ibb.co/VN9jxRK/mapteleportconfirm4k.png" alt="mapteleportconfirm4k" border="0"></a>

**Comment by zbee** [comment]
Yes! This should give me all that I need.
I will try to implement this in the next few days, and give you a build to test with.

**Comment by NoNameTrademark** [comment]
If you need other assets, ask me.

**Comment by zbee** [comment]
I was hoping to just immediately see a problem to be addressed with your resolution, but unfortunately I do not.

I tweaked the color difference tolerances to include your colors, and I did fix the location checking for the button that closes the map (which is how the script checks if the map is open).
I assume one of these was at fault, but am not certain of it.
Additionally,I also added support for checking for Chat on the left side of the screen, though this should not have impeded the functionality for you.
 
Here is a build with the tweaks, please give it a go:
[always_gatehall.zip](https://github.com/zbee/d4-always_gatehall/files/14800917/always_gatehall.zip)

---

If this does not work, I will need:
- a screenshot from you inside a dungeon with your map open (such that the map selector in the bottom left shows the "1"/"2" etc symbols on it).
- a list of what the script does in your game when you run it, similar to the [Functionality Section of the readme](https://github.com/zbee/d4-always_gatehall#functionality), so that I can understand where the script is breaking.

**Comment by zbee** [comment]
@NoNameTrademark Just notifying you of the build that is available for this issue, I know I updated this early a few days ago so it may have been easy to miss.

**Comment by NoNameTrademark** [comment]
Sorry, I've already tried them and it still doesn't work. When I use the bind 'G' it opens the maps and selects a WP which is not the correct one. Also on the 'TELEPORT CONFIRMATION' UI it seems to stop the script and does not accept the TP.

If I change my resolution to 1080 or 2k your script works. So I think it's not due to a calibration of my colors or other D4 configs.

**Comment by zbee** [comment]
Okay, so it should be turning on the map filters to disable other waypoints.
When you have your map open, it tells you to press `F` to open filters, correct? And the script does *not* open your filters at all, correct?
I will play with the accept dialog portion of the script to see what can be done there to match yours.

**Comment by NoNameTrademark** [comment]
But the UI will change in season 5, So better to will try to find the problem at that moment here you count the adapted.

**Comment by NoNameTrademark** [comment]
But the menu filters whether it is already open or closed, your script closes it and reopens it. It doesn't seem to interact with it. In the filters I still have the status of lilith disabled and when I activate them and launch your script, it no longer even clicks on a WP.

**Comment by zbee** [comment]
Ah it sounds to me like your filters menu is already open on the map when you run the script, and I don't have code to account for that, is that correct? (It is supposed to blindly open the filters menu, then disable waypoints, then close the filters)
Can you try making sure your filters menu is closed and then run the script?

I'll add logic to the filter-opening, if I understood correctly, and as I already said will work on the accept dialog to accommodate how it appears for you.

**Comment by NoNameTrademark** [comment]
The mouse always goes to the top left corner after opening the filter and always seems to select the WP in the top left quarter of my screen. No matter where I am.

For the filter your script close and repone the filter menu It doesn't matter whether it's open or closed. In the end he leaves it open

**Comment by zbee** [comment]
Right, it's searching from top left towards the bottom right for any waypoint - the clever bit is that if you turn off the waypoints in the filter menu then the Gatehall is still shown on the map.

Interesting ... Okay. It sounds like the script is failing to find the filter checkbox texture and the accept dialog textures.
I will update how both of these are searched for, probably tomorrow, and get you another build to test with.

**Comment by NoNameTrademark** [comment]
I have no option to deactivate WP

<img width="341" alt="75458" src="https://github.com/zbee/d4-always_gatehall/assets/52421870/d094cf01-2435-42fc-a101-83deddbaad1c">

**Comment by NoNameTrademark** [comment]
<img width="341" alt="75458" src="https://github.com/zbee/d4-always_gatehall/assets/52421870/d094cf01-2435-42fc-a101-83deddbaad1c">

**Comment by zbee** [comment]
Oh, and neither do I. I guess that was removed after I stopped playing for the season ... makes sense, to be fair.
I will rewrite the script tomorrow to search for the Gatehall's unique icon, and to properly click the accept dialog.
